core/services/gateway/gateway.go

```diff
@@ -83,29 +81,29 @@ func NewGatewayFromConfig(config *config.GatewayConfig, handlerFactory HandlerFa
 				return nil, fmt.Errorf("invalid node address %s", nodeConfig.Address)
 			}
 		}
-		if donConfig.HandlerName == "vault" {
-			services[donConfig.HandlerName] = vault.New(donConfig.HandlerConfig, &donConfig, donConnMgr, lggr)
-		} else {
-			handler, err := handlerFactory.NewHandler(donConfig.HandlerName, donConfig.HandlerConfig, &donConfig, donConnMgr)
-			if err != nil {
-				return nil, err
-			}
-			handlerMap[donConfig.DonId] = handler
-			donConnMgr.SetHandler(handler)
+
+		handler, err := handlerFactory.NewHandler(donConfig.HandlerName, donConfig.HandlerConfig, &donConfig, donConnMgr)
+		if err != nil {
+			return nil, err
 		}
+		handlerMap[donConfig.DonId] = handler
```

Comment:
So we'll be forced to define a DON ID for Vault DON in the same way we did it for legacy handlers, right? But this time it will be a service name? I think that should be OK... we didn't impose any restrictions on DON names - they are human-readable strings. @cedric-cordenier , @DeividasK - thoughts?

Reply:
Now we are using a single gateway.
It just sends the same request to all handlers.
Handlers must all support the single Handler interface.